.github/workflows/integration_tests.yml

```diff
@@ -0,0 +1,25 @@
+on: workflow_dispatch
+
+name: Integration tests
+
+jobs:
+  test:
+    name: Integration tests
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@master
+
+      - name: Set up JDK 1.8
+        uses: actions/setup-java@v1
+        with:
+          java-version: 1.8
+
+      - name: Decrypt secrets
+        run: integration-tests/decrypt-secrets.sh
+        env:
+          IT_SECRET_PASSPHRASE: ${{ secrets.IT_SECRET_PASSPHRASE }}
+
+      - name: Test
+        run: |
+          sbt integrationTests/test
```

Comment:
This way you will only check few simple scenarios. Instead, it should spin up an EC2 instance which will be a Test Manager. That instance should run different scenarios, verify them and verify metrics periodically.

Reply:
Verifying metrics continuously, with alerts is grafana's job. Executing the scenarios can be run on CI, I don't see why dedicated EC2 instance would be needed for that.

Reply:
CI cannot run it correctly because tests can last longer. Grafana with prometheus is only a tool for aggregating metrics. It will not show you if the test passed or now, it will not aggregate results and combine them into a summary

Reply:
The default timeout on github job is 6 hours according to [this docs](https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-syntax-for-github-actions#jobsjob_idtimeout-minutes). I don't think there's a need for any scenarios that will run longer than that and if there will be such a need I would consider alternatives only then. Regarding metrics, we can simply query for metrics during the test-run.

Reply:
we would have to pay more for such long timings on github rather than t2.micro on aws

Reply:
Ok, so let's calculate the price to make the right decision. Keep in mind that `t2.micro` might not be enough, it has only 1 GB of ram and 1 cpu. GitHub runners have [2 cpus and 7 GB of ram](https://docs.github.com/en/free-pro-team@latest/actions/reference/specifications-for-github-hosted-runners#supported-runners-and-hardware-resources).

Let say we want to post 10k transactions to a 3 node cluster. The throughput should be 50 txs * 3 nodes * 2 consensus rounds per minute, it will be 300 txs per minute, so 33 minutes for 10k txs. Running it 4 times a day Mon-Fri will cost us around 2,7k minutes of execution time per month. With the [GitHub team plan](https://docs.github.com/en/free-pro-team@latest/github/setting-up-and-managing-billing-and-payments-on-github/about-billing-for-github-actions) we have 3k minutes for free and any additional minute cost $0.008. So even if all the other workloads like compile & test on each PR would take 1k minutes a month, it would be still cheaper than the cheapest `t2.micro` which is 9.50$ per month.

So both options are rather inexpensive, but with GitHub actions we have:
- more powerful instances (more cpu and ram, which may be handy for load tests)
- we don't need to manage another piece of infrastructure, that needs to be maintained, secured, etc.

Reply:
Resolved, we leave GitHub CI for now.